recipes/quickstart/agents/Agents_101/Tool_Calling_201.ipynb

```diff
@@ -0,0 +1,774 @@
+     ]
+    }
+   ],
+   "source": [
+    "print(output)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Extracting Arxiv IDs: \n",
+    "\n",
+    "At this point, you would know the author is allergic to writing regex. To deal with this, we will simply use an `8b` instance to extract the `arxiv id` from the paper:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_arxiv_ids(web_results: dict, temperature: int = 0, max_tokens=512):\n",
+    "    # Initialize chat history with a specific prompt to extract arXiv IDs\n",
+    "    arxiv_id_chat_history = [{\"role\": \"system\", \"content\": \"Given this input, give me the arXiv ID of the papers. The input has the query and web results. DO NOT WRITE ANYTHING ELSE IN YOUR RESPONSE: ONLY THE ARXIV ID ONCE, the web search will have it repeated mutliple times, just return the it once and where its actually the arxiv ID\"}, {\"role\": \"user\", \"content\": f\"Here is the query and results{web_results}\"}]\n",
+    "\n",
+    "    # Call the model to process the input and extract arXiv IDs\n",
+    "    response = client.chat.completions.create(\n",
+    "        model=\"llama-3.1-8b-instant\",  # Adjust the model as necessary\n",
+    "        messages=arxiv_id_chat_history,\n",
+    "        max_tokens=max_tokens,\n",
+    "        temperature=temperature\n",
+    "    )\n",
+    "    \n",
+    "    # Append the assistant's response to the chat history\n",
+    "    arxiv_id_chat_history.append({\n",
+    "        \"role\": \"assistant\",\n",
+    "        \"content\": response.choices[0].message.content\n",
+    "    })\n",
+    "    \n",
+    "    # Return the extracted arXiv IDs\n",
+    "    return response.choices[0].message.content"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2407.21783\n",
+      "2103.11943\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(get_arxiv_ids(search_results[0]))\n",
+    "print(get_arxiv_ids(search_results[1]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Downloading the papers and extracting details: \n",
+    "\n",
+    "3.1 family LLM(s) are great enough to use raw outputs extracted from a PDF and summarise them. However, we are still bound by their (great) 128k context length-to live with this, we will extract just the first 80k words. \n",
```

Comment:
```suggestion
    "Llama 3.1 family LLM(s) are great enough to use raw outputs extracted from a PDF and summarise them. However, we are still bound by their (great) 128k context length-to live with this, we will extract just the first 80k words. \n",
```

Reply:
how about if we add the 3.2 OCR capabilities here following with passing two pdfs to compare them as well?